drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1800 @@
+    return
+  end
+
+  -- @field public byte_length number 1
+  -- @field public SUCCESS number 0
+  -- @field public FAILURE number 1
+  -- @field public DUPLICATE number 2
+  -- @field public OCCUPIED number 3
+  -- @field public INVALID_FIELD number 133
+  -- @field public RESOURCE_EXHAUSTED number 137
+  -- @field public NOT_FOUND number 139
+ 
+  -- Update commandResult
+  status = "occupied"
+  if elements.status.value == DoorLock.types.DlStatus.FAILURE then
+    status = "failure"
+  elseif elements.status.value == DoorLock.types.DlStatus.DUPLICATE then
+    status = "duplicate"
+  elseif elements.status.value == DoorLock.types.DlStatus.INVALID_FIELD then
+    status = "invalidCommand"
+  elseif elements.status.value == DoorLock.types.DlStatus.RESOURCE_EXHAUSTED then
+    status = "resourceExhausted"
+  elseif elements.status.value == DoorLock.types.DlStatus.NOT_FOUND then
+    status = "failure"
+  end
+  log.info_with({hub_logs=true}, string.format("Result: %s", status)) -- needs to remove
+
+  if status ~= "occupied" then
+    local result = {
+      commandName = cmdName,
+      statusCode = status
+    }
+    local event = capabilities.lockCredentials.commandResult(
+      result,
+      {
+        state_change = true,
+        visibility = {displayed = false}
+      }
+    )
+    device:emit_event(event)
+    device:set_field(lock_utils.BUSY_STATE, false, {persist = true})
+    return
+  end
+
+  if elements.next_credential_index.value ~= nil then
+    -- Get parameters
+    local credIdx = elements.next_credential_index.value
+    local credential = {
+      credential_type = DoorLock.types.DlCredentialType.PIN,
+      credential_index = credIdx,
+    }
+    local credData = device:get_field(lock_utils.CRED_DATA)
+    local userIdx = device:get_field(lock_utils.USER_INDEX)
+    local userType = device:get_field(lock_utils.USER_TYPE)
+    local userTypeMatter = DoorLock.types.UserTypeEnum.UNRESTRICTED_USER
+    if userType == "guest" then
+      userTypeMatter = DoorLock.types.UserTypeEnum.SCHEDULE_RESTRICTED_USER
+    end
+
+    -- needs to remove logs
+    log.info_with({hub_logs=true}, string.format("credentialIndex: %s", credIdx))
+    log.info_with({hub_logs=true}, string.format("credData: %s", credData))
+    log.info_with({hub_logs=true}, string.format("userIndex: %s", userIdx))
+    log.info_with({hub_logs=true}, string.format("userType: %s", userType))
+
+    device:set_field(lock_utils.CRED_INDEX, credIdx, {persist = true})
+
+    -- Sned command
+    local ep = find_default_endpoint(device, DoorLock.ID)
+    device:send(
+      DoorLock.server.commands.SetCredential(
+        device, ep,
+        DoorLock.types.DlDataOperationType.ADD, -- Data Operation Type: Add(0), Modify(2)
+        credential,    -- Credential
+        credData,      -- Credential Data
+        userIdx,       -- User Index
+        nil,           -- User Status
+        userTypeMatter -- User Type
```

Comment:
In my testing we're hitting this for a device that requires COTA pins and it's using `UserTypeEnum.UNRESTRICTED_USER` when I think it should be using `UserTypeEnum.REMOTE_ONLY_USER`